atomic_reactor/tasks/common.py

```diff
@@ -107,3 +112,31 @@ def load_workflow_data(self) -> inner.ImageBuildWorkflowData:
 
     def load_config(self) -> config.Configuration:
         return config.Configuration(self._params.config_file)
+
+    def throw_canceled_build_exception(self):
+        self.data.build_canceled = True
+        raise BuildCanceledException("Build was canceled")
+
+    @classmethod
+    def run(cls, init_build_dirs: Optional[bool] = False):
+        try:
+            cls.fs_watcher.start()
+
+            # non-plugin based tasks don't have the attribute
+            if hasattr(cls, 'keep_plugins_running') and cls.keep_plugins_running:
```

Comment:
Maybe we could move this to Task level and rename? I can only think of silly names though, like `immortal` 😄

Reply:
Well we're using it here :smile: 

I imagine that if we reworked the exit task to not be plugin based, we would still want to ignore SIGTERM in it

Reply:
so what about `keep_task_running` or `cancellable_task`

Reply:
`ignore_sigterm` would be a good name if it's just for ignoring SIGTERM

Come to think of it, maybe it *should* be just for that, and `keep_plugins_running` should be a separate thing that decides only if plugin errors will be ignored

Reply:
added `ignore_sigterm` class variable, since it should do different thing than `keep_plugins_running`